tests/beaker_tests/netdev_stdlib/port_channel_provider_defaults.rb

```diff
@@ -0,0 +1,230 @@
+#   :title_pattern will be set to 'id'.
+#
+
+tests['default_properties'] = {
+  title_pattern:  'port-channel100',
+  manifest_props: "
+    id            => '100',
+    interfaces    => ['ethernet1/8'],
+    minimum_links => '1',
+  ",
+  code:           [0, 2],
+  resource_props: {
+    'id'            => '100',
+    'interfaces'    => "['ethernet1/8']",
+    'minimum_links' => '1',
+  },
+}
+
+tests['non_default_properties'] = {
+  title_pattern:  'port-channel100',
+  manifest_props: "
+    id            => '100',
+    interfaces    => ['ethernet1/8', 'ethernet1/9'],
+    minimum_links => '3',
+  ",
+  resource_props: {
+    'id'            => '100',
+    'interfaces'    => "['ethernet1/8', 'ethernet1/9']",
+    'minimum_links' => '3',
+  },
+}
+
+#################################################################
+# HELPER FUNCTIONS
+#################################################################
+
+# Full command string for puppet resource command
+def puppet_resource_cmd
+  cmd = PUPPET_BINPATH +
+        'resource port_channel port-channel100'
+  get_namespace_cmd(agent, cmd, options)
+end
+
+def build_manifest_portchannel(tests, id)
+  if tests[id][:ensure] == :absent
+    state = 'ensure => absent,'
+    tests[id][:resource] = {}
+  else
+    state = 'ensure => present,'
+    manifest = tests[id][:manifest_props]
+    tests[id][:resource] = tests[id][:resource_props]
+  end
+
+  tests[id][:title_pattern] = id if tests[id][:title_pattern].nil?
+  logger.debug(
+    "build_manifest_portchannel :: title_pattern:\n" +
+             tests[id][:title_pattern])
+  tests[id][:manifest] = "cat <<EOF >#{PUPPETMASTER_MANIFESTPATH}
+  node 'default' {
+    port_channel { 'port-channel100':
+      #{state}
+      #{manifest}
+    }
+  }
+EOF"
+end
+
+def test_harness_portchannel(tests, id)
+  tests[id][:ensure] = :present if tests[id][:ensure].nil?
+  tests[id][:resource_cmd] = puppet_resource_cmd
+  tests[id][:desc] += " [ensure => #{tests[id][:ensure]}]"
+
+  # Build the manifest for this test
+  build_manifest_portchannel(tests, id)
+
+  # test_harness_common(tests, id)
+  test_manifest(tests, id)
+  test_resource(tests, id)
+  test_idempotence(tests, id)
+
+  tests[id][:ensure] = nil
+end
+
+#################################################################
+# TEST CASE EXECUTION
+#################################################################
+test_name "TestCase :: #{testheader}" do
+  # -------------------------------------------------------------------
+  device = platform
+  logger.info("#### This device is of type: #{device} #####")
+  resource_absent_cleanup(agent, 'port_channel',
+                          'Setup switch for port_channel provider test')
+  logger.info("\n#{'-' * 60}\nSection 1. Default Property Testing")
+
+  case device
+  when /n7k/
+    id = 'default_properties'
+  when /n5k|n6k/
+    id = 'default_properties'
+  when /n3k|n9k/
+    id = 'default_properties'
+  end
```

Comment:
We have a more streamlined way to handle this now.  Please use the `platform:` hash key instead.

https://github.com/cisco/cisco-network-puppet-module/blob/release_1.2.0/tests/beaker_tests/cisco_command_config/test_command_config.rb#L97